osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -326,7 +326,7 @@ public override bool OnExiting(IScreen next)
         {
             // the room may not be left immediately after a disconnection due to async flow,
             // so checking the IsConnected status is also required.
-            if (client.Room == null || !client.IsConnected.Value)
+            if (client?.Room == null || !client.IsConnected.Value)
```

Comment:
How does this one occur? Can `OnExiting` somehow be called before a load finishes? This seems to also be the same code in `master`.

Reply:
This was added in passing due to one of the tests I added, but can be reverted and done as a separate PR. It is as you describe - screens exited during async load.

Can be reverted because this would still not really fix the issue as it would then go to `base.OnExiting()` where it tries to transform and dies on trying to apply transforms during async load (o!f exception).

Reply:
Can be tested with:
```diff
diff --git a/osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayer.cs b/osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayer.cs
index 92f9c5733f..5a1f74fca2 100644
--- a/osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayer.cs
+++ b/osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayer.cs
@@ -126,9 +126,6 @@ public void TestJoinRoomWithoutPassword()
             AddStep("refresh rooms", () => multiplayerScreen.RoomManager.Filter.Value = new FilterCriteria());
             AddStep("select room", () => InputManager.Key(Key.Down));
             AddStep("join room", () => InputManager.Key(Key.Enter));
-
-            AddUntilStep("wait for room open", () => this.ChildrenOfType<MultiplayerMatchSubScreen>().FirstOrDefault()?.IsLoaded == true);
-            AddUntilStep("wait for join", () => client.Room != null);
         }
 
         [Test]
@@ -179,9 +176,6 @@ public void TestJoinRoomWithPassword()
             AddUntilStep("password prompt appeared", () => (passwordEntryPopover = InputManager.ChildrenOfType<DrawableRoom.PasswordEntryPopover>().FirstOrDefault()) != null);
             AddStep("enter password in text box", () => passwordEntryPopover.ChildrenOfType<TextBox>().First().Text = "password");
             AddStep("press join room button", () => passwordEntryPopover.ChildrenOfType<OsuButton>().First().Click());
-
-            AddUntilStep("wait for room open", () => this.ChildrenOfType<MultiplayerMatchSubScreen>().FirstOrDefault()?.IsLoaded == true);
-            AddUntilStep("wait for join", () => client.Room != null);
         }
 
         [Test]
diff --git a/osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs b/osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs
index f35671a761..561fa220c8 100644
--- a/osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs
+++ b/osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs
@@ -326,7 +326,7 @@ public override bool OnExiting(IScreen next)
         {
             // the room may not be left immediately after a disconnection due to async flow,
             // so checking the IsConnected status is also required.
-            if (client?.Room == null || !client.IsConnected.Value)
+            if (client.Room == null || !client.IsConnected.Value)
             {
                 // room has not been created yet; exit immediately.
                 return base.OnExiting(next);
```
```
TearDown : System.NullReferenceException : Object reference not set to an instance of an object.
--TearDown
   at osu.Game.Screens.OnlinePlay.Multiplayer.MultiplayerMatchSubScreen.OnExiting(IScreen next) in /home/smgi/Repos/osu/osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs:line 329
   at osu.Game.Screens.OnlinePlay.OnlinePlayScreen.OnExiting(IScreen next) in /home/smgi/Repos/osu/osu.Game/Screens/OnlinePlay/OnlinePlayScreen.cs:line 243
   at osu.Framework.Screens.ScreenStack.exitFrom(IScreen source, Boolean shouldFireExitEvent, Boolean shouldFireResumeEvent)
   at osu.Framework.Screens.ScreenStack.Exit(IScreen source)
   at osu.Framework.Screens.ScreenStack.Exit()
   at osu.Game.Tests.Visual.ScreenTestScene.<addExitAllScreensStep>b__12_0() in /home/smgi/Repos/osu/osu.Game/Tests/Visual/ScreenTestScene.cs:line 51
   at osu.Framework.Testing.Drawables.Steps.UntilStepButton.<>c__DisplayClass11_0.<.ctor>b__0()
   at osu.Framework.Testing.Drawables.Steps.StepButton.PerformStep(Boolean userTriggered)
   at osu.Framework.Testing.TestScene.runNextStep(Action onCompletion, Action`1 onError, Func`2 stopCondition)
--- End of stack trace from previous location ---
   at osu.Framework.Testing.TestSceneTestRunner.TestRunner.RunTestBlocking(TestScene test)
   at osu.Game.Tests.Visual.OsuTestScene.OsuTestSceneTestRunner.RunTestBlocking(TestScene test) in /home/smgi/Repos/osu/osu.Game/Tests/Visual/OsuTestScene.cs:line 371
   at osu.Framework.Testing.TestScene.RunTests()
```

Reply:
Will revert for now.